src/onnx/parse_instancenorm.cpp

```diff
@@ -60,32 +60,69 @@ struct parse_instancenorm : op_parser<parse_instancenorm>
             MIGRAPHX_THROW(opd.op_name + ": invalid output type: " + std::to_string(dtype) +
                            ". Valid types are 1 (float), 10 (half), and 11 (double).");
 
-        auto ndims = dims.size();
+        bool dyn_input = x->get_shape().dynamic();
+        auto ndims = x->get_shape().ndim();
         assert(ndims >= 2);
         auto kdims = ndims - 2;
+                                                               
+        if(x->get_shape().dynamic())
+        {
+            dims= x->get_shape().min_lens();
+        }
 
         std::vector<int64_t> axes(kdims);
         std::iota(axes.begin(), axes.end(), 2);
 
         auto mean = info.add_instruction(make_op("reduce_mean", {{"axes", axes}}), x);
-        auto mean_bcast =
-            info.add_instruction(make_op("multibroadcast", {{"out_lens", dims}}), mean);
+        // Create a multibroadcast instruction.  Use the 2-argument
+        // version of the instruction for dynamic input and the 1-argument version
+        // for static input, since future compiler passes may optimize them differently.
+        instruction_ref mean_bcast;
+        if(dyn_input)
+            mean_bcast =
+                info.add_instruction(make_op("multibroadcast"), mean, x);
+        else
+            mean_bcast =
+                info.add_instruction(make_op("multibroadcast", {{"out_lens", dims}}), mean);
```

Comment:
This should just use `add_common_op` to insert the broadcasts instead of manually inserting them.

Reply:
What's the difference?  I see a commit comment about implicit type conversion for broadcastable ops--is there a functional difference?

Reply:
done.  We agreed offline that `add_common_op()` can't be used with the broadcast op but I implemented it for multibroadcast.